metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -40,7 +40,21 @@ def ingest() -> None:
     help="Config file in .toml or .yaml format.",
     required=True,
 )
-def run(config: str) -> None:
+@click.option(
+    "--dry-run",
+    type=bool,
+    is_flag=True,
+    default=False,
+    help="Perform a dry run of the ingestion, essentially skipping writing to sink.",
+)
+@click.option(
+    "--preview-mode",
+    type=bool,
+    is_flag=True,
+    default=False,
+    help="Perform limited ingestion from the source to the sink to get a quick preview.",
+)
+def run(config: str, dry_run: bool, preview_mode: bool) -> None:
```

Comment:
if you change the option name from preview_mode to preview, you might have to change this variable name as well